lib/process-services-cloud/src/lib/group/components/group-cloud.component.ts

```diff
@@ -0,0 +1,255 @@
+import { Observable, of, BehaviorSubject } from 'rxjs';
+import { GroupModel, GroupSearchParam } from '../models/group.model';
+import { GroupCloudService } from '../services/group-cloud.service';
+import { debounceTime } from 'rxjs/internal/operators/debounceTime';
+import { distinctUntilChanged, switchMap, flatMap, mergeMap, filter, tap } from 'rxjs/operators';
+
+@Component({
+    selector: 'adf-cloud-group',
+    templateUrl: './group-cloud.component.html',
+    styleUrls: ['./group-cloud.component.scss'],
+    animations: [
+        trigger('transitionMessages', [
+            state('enter', style({ opacity: 1, transform: 'translateY(0%)' })),
+            transition('void => enter', [
+                style({ opacity: 0, transform: 'translateY(-100%)' }),
+                animate('300ms cubic-bezier(0.55, 0, 0.55, 0.2)')
+            ])
+        ])
+    ],
+    encapsulation: ViewEncapsulation.None
+})
+export class GroupCloudComponent implements OnInit {
+
+    static MODE_SINGLE = 'single';
+    static MODE_MULTIPLE = 'multiple';
+    static NO_MATCH_FOUND = 'match';
+    static NO_ROLE_MAPPING = 'rolemapping';
+
+    @ViewChild('groupInput') groupInput: ElementRef<HTMLInputElement>;
+
+    /** (required) Name of the app. */
+    @Input()
+    applicationName: string;
+
+    /** Mode of the group selection (single/multiple) */
+    @Input()
+    mode: string = GroupCloudComponent.MODE_SINGLE;
+
+    @Input()
+    preSelectGroups: GroupModel[] = [];
+
+    /** Emitted when a group is selected. */
+    @Output()
+    selectGroup: EventEmitter<GroupModel> = new EventEmitter<GroupModel>();
+
+    /** Emitted when a group is removed. */
+    @Output()
+    removeGroup: EventEmitter<GroupModel> = new EventEmitter<GroupModel>();
+
+    private selectedGroups: GroupModel[] = [];
+
+    private searchGroups: GroupModel[] = [];
+
+    private searchGroupsSubject: BehaviorSubject<GroupModel[]>;
+
+    private selectedGroupsSubject: BehaviorSubject<GroupModel[]>;
+
+    searchGroups$: Observable<GroupModel[]>;
+
+    selectedGroups$: Observable<GroupModel[]>;
+
+    searchGroupsControl: FormControl = new FormControl();
+
+    _subscriptAnimationState = 'enter';
+
+    applicationId: string;
+
+    searchedValue = '';
+
+    constructor(private groupService: GroupCloudService) {
+        this.selectedGroupsSubject = new BehaviorSubject<GroupModel[]>(this.selectedGroups);
+        this.searchGroupsSubject = new BehaviorSubject<GroupModel[]>(this.searchGroups);
+        this.selectedGroups$ = this.selectedGroupsSubject.asObservable();
+        this.searchGroups$ = this.searchGroupsSubject.asObservable();
+    }
+
+    ngOnInit() {
+        this.loadPreSelectGroups();
+        this.initSearch();
+        if (this.applicationName) {
+            this.loadClientId();
+        }
+    }
+
+    private async loadClientId() {
+        this.applicationId = await this.groupService.getClientId(this.applicationName);
```

Comment:
Disable the search till the loadClientId doesn't return

Reply:
not sure async await does actually wait as it is in the ngoninit

Reply:
it doesn't wait but until then the search field is disabled